File: GroceryTracker.Backend/Model/Dto/SearchResultsDto.cs

using System.Collections.Generic;
using GroceryTracker.Backend.DatabaseAccess.Enum;
using GroceryTracker.Backend.Search;

namespace GroceryTracker.Backend.Model.Dto
{
   public class SearchResultsDto
   {
      public ArticleSearch Search { get; set; }
      public IEnumerable<SearchResultDto> Results { get; set; }
   }

   public class SearchResultDto
   {
      public int Id { get; set; }

      public string ArticleName { get; set; }

      public string Details { get; set; }

      public string BrandName { get; set; }

      public string CategoryName { get; set; }

      public SearchMatchType ArticleMatch { get; set; }

      public SearchMatchType DetailsMatch { get; set; }

      public SearchMatchType BrandMatch { get; set; }

      public SearchMatchType CategoryMatch { get; set; }

      public double LastPrice { get; set; }

      public double AveragePrice { get; set; }
   }
}

File: GroceryTracker.Frontend/src/dtos/searchResultsDto.ts
export interface SearchResultsDto {
   search: Search
   results: SearchResultDto[]
}

export interface Search {
   originalSearchString: string
   articleName: string
   brandName: string
   primaryCategory: string
   details: string
   dynamicSearchString: string
   resultLimit: 10
}

export interface SearchResultDto {
   articleName: string
   brandName: string
   details: string
   categoryName: string
   articleMatch: SearchMatchType
   detailsMatch: SearchMatchType
   brandMatch: SearchMatchType
   categoryMatch: SearchMatchType
   lastPrice: number
   averagePrice: number
}

export enum SearchMatchType {
   Query,
   Dynamic,
   NoMatch,
}
